pkg/wallet/internal/mapping/mapping_list_actions.go

```diff
@@ -34,68 +35,77 @@ func MapListActionsArgs(args sdk.ListActionsArgs) wdk.ListActionsArgs {
+		result.IncludeInputs = to.Ptr(primitives.BooleanDefaultFalse(*args.IncludeInputs))
 	}
 
 	if args.IncludeOutputs != nil {
-		includeOutputs := primitives.BooleanDefaultFalse(*args.IncludeOutputs)
-		result.IncludeOutputs = &includeOutputs
+		result.IncludeOutputs = to.Ptr(primitives.BooleanDefaultFalse(*args.IncludeOutputs))
 	}
 
 	if args.IncludeLabels != nil {
-		includeLabels := primitives.BooleanDefaultFalse(*args.IncludeLabels)
-		result.IncludeLabels = &includeLabels
+		result.IncludeLabels = to.Ptr(primitives.BooleanDefaultFalse(*args.IncludeLabels))
 	}
 
 	if args.IncludeInputSourceLockingScripts != nil {
-		includeInputSourceLockingScripts := primitives.BooleanDefaultFalse(*args.IncludeInputSourceLockingScripts)
-		result.IncludeInputSourceLockingScripts = &includeInputSourceLockingScripts
+		result.IncludeInputSourceLockingScripts = to.Ptr(primitives.BooleanDefaultFalse(*args.IncludeInputSourceLockingScripts))
 	}
 
 	if args.IncludeInputUnlockingScripts != nil {
-		includeInputUnlockingScripts := primitives.BooleanDefaultFalse(*args.IncludeInputUnlockingScripts)
-		result.IncludeInputUnlockingScripts = &includeInputUnlockingScripts
+		result.IncludeInputUnlockingScripts = to.Ptr(primitives.BooleanDefaultFalse(*args.IncludeInputUnlockingScripts))
 	}
 
 	if args.IncludeOutputLockingScripts != nil {
-		includeOutputLockingScripts := primitives.BooleanDefaultFalse(*args.IncludeOutputLockingScripts)
-		result.IncludeOutputLockingScripts = &includeOutputLockingScripts
+		result.IncludeOutputLockingScripts = to.Ptr(primitives.BooleanDefaultFalse(*args.IncludeOutputLockingScripts))
 	}
 
 	return result
 }
 
 // MapListActionsResult maps *wdk.ListActionsResult to *sdk.ListActionsResult
-func MapListActionsResult(result *wdk.ListActionsResult) *sdk.ListActionsResult {
-	return &sdk.ListActionsResult{
-		TotalActions: result.TotalActions,
-		Actions:      slices.Map(result.Actions, mapListActionsAction),
+func MapListActionsResult(result *wdk.ListActionsResult) (*sdk.ListActionsResult, error) {
+	actions, err := slices.MapOrError(result.Actions, mapListActionsAction)
+	if err != nil {
+		return nil, fmt.Errorf("failed to map actions: %w", err)
+	}
+
+	if result.TotalActions > math.MaxUint32 {
+		return nil, fmt.Errorf("total actions count %d exceeds uint32 maximum", result.TotalActions)
 	}
+
+	return &sdk.ListActionsResult{
+		TotalActions: uint32(result.TotalActions),
+		Actions:      actions,
+	}, nil
 }
 
 // mapListActionsAction maps wdk.WalletAction to sdk.Action
-func mapListActionsAction(action wdk.WalletAction) sdk.Action {
+func mapListActionsAction(action wdk.WalletAction) (sdk.Action, error) {
 	var txidHash chainhash.Hash
-	if action.TxID != "" {
-		if hash, err := chainhash.NewHashFromHex(action.TxID); err == nil {
-			txidHash = *hash
-		}
+
+	hash, err := chainhash.NewHashFromHex(action.TxID)
+	if err != nil {
+		return sdk.Action{}, fmt.Errorf("failed to convert txid to hash: %w", err)
+	}
+	txidHash = *hash
```

Comment:
no need for this variable, you can simply assign *hash to Txid field of sdk.Action